src/Helmet.js

```diff
@@ -552,8 +180,18 @@ const Helmet = (Component) => class HelmetWrapper extends React.Component {
             nestedChildren &&
             typeof nestedChildren !== "string"
         ) {
-            console.warn(`Helmet expects a single string as a child of ${child.type}`);
+            if (!VALID_TAG_NAMES.some(name => child.type === name)) {
```

Comment:
You're right. I updated the logic and added a test for self-closing invalid elements.